controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller_test.go

```diff
@@ -0,0 +1,165 @@
+package spaceprovisionerconfig
+
+import (
+	"context"
+	"errors"
+	"os"
+	"testing"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/apis"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestSpaceProvisionerConfigValidation(t *testing.T) {
+	t.Run("is not valid when non-existing ToolchainCluster is referenced", func(t *testing.T) {
+		// given
+		spc := &toolchainv1alpha1.SpaceProvisionerConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "spc",
+				Namespace: test.HostOperatorNs,
+			},
+			Spec: toolchainv1alpha1.SpaceProvisionerConfigSpec{
+				ToolchainCluster: "non-existent",
+			},
+		}
+		r, req, cl := prepareReconcile(t, spc)
+
+		// when
+		_, reconcileErr := r.Reconcile(context.TODO(), req)
+		getErr := cl.Get(context.TODO(), runtimeclient.ObjectKey{Name: "spc", Namespace: test.HostOperatorNs}, spc)
+
+		// then
+		require.NoError(t, reconcileErr)
+		require.NoError(t, getErr)
+		assert.True(t, condition.IsFalse(spc.Status.Conditions, toolchainv1alpha1.ConditionReady))
+	})
+
+	t.Run("is valid when existing ToolchainCluster is referenced", func(t *testing.T) {
+		// given
+		spc := &toolchainv1alpha1.SpaceProvisionerConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "spc",
+				Namespace: test.HostOperatorNs,
+			},
+			Spec: toolchainv1alpha1.SpaceProvisionerConfigSpec{
+				ToolchainCluster: "cluster1",
+			},
+		}
+		r, req, cl := prepareReconcile(t, spc, &toolchainv1alpha1.ToolchainCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cluster1",
+				Namespace: test.HostOperatorNs,
+			},
+		})
+
+		// when
+		_, reconcileErr := r.Reconcile(context.TODO(), req)
+		getErr := cl.Get(context.TODO(), runtimeclient.ObjectKey{Name: "spc", Namespace: test.HostOperatorNs}, spc)
```

Comment:
the get calls can be simplified
```suggestion
		getErr := cl.Get(context.TODO(), runtimeclient.ObjectKeyFromObject(spc), spc)
```